src/Schema/EnumSchema.php

```diff
@@ -29,6 +30,37 @@ public function toArray(): array
         return [
             'description' => $this->description,
             'enum' => $this->options,
+            'type' => $this->getOptionTypes(),
         ];
     }
+
+    /**
+     * @return string|array<int, string>
+     */
+    protected function getOptionTypes(): array|string
+    {
+        $types = [];
+        foreach ($this->options as $option) {
+            $type = gettype($option);
+            $types[] = match ($type) {
+                'integer', 'double' => 'number',
+                'string' => 'string'
+            };
+        }
+
+        if ($this->nullable) {
+            $types[] = 'null';
+        }
+
+        $types = collect($types)
+            ->unique()
+            ->values()
+            ->all();
+
+        if (count($types) === 1) {
+            return $types[0];
+        }
+
+        return $types;
+    }
```

Comment:
This feels a little bit ugly. Happy to take any suggestions you may have!

Reply:
Maybe it would just be cleaner if we make it a single collection pipeline?